test/rules/restrict-plus-operands/test.ts.lint

```diff
@@ -17,35 +17,38 @@ var pair: NumberStringPair = {
 
 // bad
 var bad1  = 5 + "10";
-            ~~~~~~~~  [Operands of '+' operation must either be both strings or both numbers, consider using template literals]
+            ~~~~~~~~  [Operands of '+' operation must either be both strings or both numbers, but found 5 + "10". Consider using template literals.]
 var bad2  = [] + 5;
-            ~~~~~~  [Operands of '+' operation must either be both strings or both numbers]
+            ~~~~~~  [Operands of '+' operation must either be both strings or both numbers, but found undefined[] + 5]
```

Comment:
> `undefined[]`

Agreed, this is a little confusing (and `never[]` would be more accurate in some situations). For array literals of size 0, let's just have the rule print `[]`.

> `Operands of '+' operation must either be both strings or numbers, but found [] + {}`

Reply:
Done.